libcontainer/container_linux.go

```diff
@@ -484,6 +493,12 @@ func (c *linuxContainer) commandTemplate(p *Process, childPipe *os.File) (*exec.
 		fmt.Sprintf("_LIBCONTAINER_INITPIPE=%d", stdioFdCount+len(cmd.ExtraFiles)-1),
 		fmt.Sprintf("_LIBCONTAINER_STATEDIR=%s", c.root),
 	)
+
+	cmd.ExtraFiles = append(cmd.ExtraFiles, logPipe)
```

Comment:
This side of the pipe should be closed after start as it's provided to the new process and dup'd

Reply:
There also needs to be code in `init_linux.go` to close the pipe before we `execve` the user code.

Reply:
We already do so:
- initProcess: https://github.com/masters-of-cats/runc/blob/c486e3c40633d571df38fca56da69f3ab0ab13fe/libcontainer/process_linux.go#L291
- setnsProcess: https://github.com/masters-of-cats/runc/blob/c486e3c40633d571df38fca56da69f3ab0ab13fe/libcontainer/process_linux.go#L93

Reply:
@cyphar Isn't that done generically for all extra fds (except the ones requested by the user) as part of [finalizeNamespace(https://github.com/masters-of-cats/runc/blob/f5b99917df9fbe1d9a4114966fb088dd6860e85a/libcontainer/init_linux.go#L126) that is being called by both init and setns flavours of init lifecycle?

Reply:
Yeah, that make sense.